### ISSUE TITLE
PLW2901 ignore when using cast

### ISSUE THREAD

**Issue by LefterisJP** [charliermarsh/ruff#3652]
```
ruff 0.0.257
```

```python
for x, x_type in events:
     if x_type == 'something':
         x = cast(Event1, x)
     elif x_type == 'something_else':
         x = cast(Event2, x)
```

This hits with PLW2901. I guess it would make sense to ignore this error in case of `cast()` which is only used for typing.

**Comment by JonathanPlasse** [comment]
I would like to work on it.

**Comment by LefterisJP** [comment]
> In this case, this is not an error, you override `x` with a different variable `event`. This should be flagged. However, if you cast `x` on itself like below. It makes sense to ignore `PLW2901`.
> 
> ```python
> for x, x_type in events:
>      if x_type == 'something':
>          x = cast(Event1, x)
>      elif x_type == 'something_else':
>          x = cast(Event2, x)
> ```

Ah sorry Jonathan, I did not run it through ruff -- tried to make a pseudocode of what I seen in our code. Will edit it.

**Comment by dhruvmanila** [comment]
> I would like to work on it.

@JonathanPlasse are you working on this?
